cmd/main.go

```diff
@@ -35,9 +36,9 @@ func main() {
 			&pluginv2.Plugin{},
 		),
 		cli.WithExtraCommands(
-			newCompletionCmd(),
 			version.NewCmd(),
 		),
+		cli.WithCompletion,
```

Comment:
IMO the Completion also ought to be a plugin.

Reply:
This completion command is valid for every plugin combination you use. It is not specific to `go.kubebuilder.io` plugin. There is no need to provide a plugin that makes completion because this command will work for the plugins you configure.